grades/api.py

```diff
@@ -230,3 +232,35 @@ def generate_program_certificate(user, program):
         user.username,
         program.title
     )
+
+
+def update_or_create_combined_final_grade(user, course):
+    """
+    Update or create CombinedFinalGrade
+
+    Args:
+        user (User): a django User
+        course (Course): a course model object
+    """
+    if course.has_exam:
```

Comment:
Suggest inverting this to a `if not course.has_exam: return` so you don't have to indent the whole method.